src/NHibernate/Cache/Entry/CollectionCacheEntry.cs

```diff
@@ -31,6 +31,24 @@ public static CollectionCacheEntry Create(IPersistentCollection collection, ICol
 			};
 		}
 
+		public static bool TryCreate(IPersistentCollection collection, ICollectionPersister persister, out CollectionCacheEntry entry)
+		{
+			entry = null;
+			try
+			{
+				entry = new CollectionCacheEntry
+				{
+					state = collection.Disassemble(persister)
+				};
+
+				return true;
+			}
+			catch
```

Comment:
Having a catch-all here is really bad.

Ideally we should not have a catch here at all.